d3a_interface/utils.py

```diff
@@ -90,6 +90,19 @@ def str_to_pendulum_datetime(input_str):
     return pendulum_time
 
 
+def ui_str_to_pendulum_datetime(input_str):
+    if input_str is None:
+        return None
+    try:
+        pendulum_time = from_format(input_str, TIME_FORMAT)
+    except ValueError:
+        try:
+            pendulum_time = from_format(input_str, DATE_TIME_UI_FORMAT)
+        except ValueError:
+            raise Exception(f"Format is not one of ('{TIME_FORMAT}', '{DATE_TIME_FORMAT}')")
+    return pendulum_time
+
+
```

Comment:
Why not adding 
```
        try:
            pendulum_time = from_format(input_str, DATE_TIME_UI_FORMAT)
        except ValueError:
```
to str_to_pendulum_datetime and reuse it?

Reply:
`str_to_pendulum_datetime ` also being used in `convert_str_to_pendulum_in_dict`

Reply:
hmm, doesn't sell it to me, sorry. 
Sorry to be relentless here, but a function name should explain what it does. And `ui_str_to_pendulum_datetime` predicts that the only handles ui-strings. But why not enabling `str_to_pendulum_datetime` to handle ui_strings?

Reply:
> Sorry to be relentless here, but a function name should explain what it does

I have used `ui` as prefix, so Im not sure how much more clarity you need to explain its been used i ui.

> And ui_str_to_pendulum_datetime predicts that the only handles ui-strings.

Yes!

> But why not enabling str_to_pendulum_datetime to handle ui_strings?

Because `str_to_pendulum_datetime ` is used in d3a.

Reply:
Discussed offline, `ui_str_to_pendulum_datetime`will only handles ui datetime string.